testing/test_config.py

```diff
@@ -352,6 +352,25 @@ def test_inifilename(self, tmpdir):
         assert config.inicfg.get('name') == 'value'
         assert config.inicfg.get('should_not_be_set') is None
 
+    @pytest.mark.issue1618
+    def test_consider_plugin(self, testdir):
+        pytest.importorskip('xdist')
```

Comment:
i am certain that xdist is currently the only thing triggering the behavior

the problem in question is the worker process ignoring the 'no:foo' plugin spec